FlowCrypt/Functionality/Mail Provider/SearchMessage Provider/GmailSearchExpressionGenerator.swift

```diff
@@ -8,15 +8,18 @@
 
 import Foundation
 
-protocol GmailSearchBackupGenerator {
-    func makeBackupQuery(with expressions: [String]) -> String
+protocol GmailBackupSearchQueryProviderType {
+    func makeBackupQuery(for email: String) -> String?
 }
 
-final class GmailSearchExpressionGenerator: GmailSearchBackupGenerator {
-    func makeBackupQuery(with expressions: [String]) -> String {
-        let folderQuery = "in:anywhere"
-        let search = expressions.map { "\"\($0)\"" }
-        let searchExpressionQuery = search.joined(separator: " OR ")
-        return folderQuery + " " + searchExpressionQuery + " has:attachment"
+final class GmailBackupSearchQueryProvider: GmailBackupSearchQueryProviderType {
+    let core: Core
+
+    init(core: Core = .shared) {
+        self.core = core
+    }
+
+    func makeBackupQuery(for email: String) -> String? {
+        core.gmailBackupSearch(for: email)
```

Comment:
I think this class became superfluous - the only actual code here is `core.gmailBackupSearch(for: email)` which could be called from one level up directly.

Reply:
True, it was done for testing purposes 
`GmailService` initialised with `GmailBackupSearchQueryProviderType` instead of being initialised with `Core` class which is a shared instance. 

What do you think of making `call` functions from `Core` class as public functions, and spread some functionality to other classes? In this case Core will be kind of interface for communicating. 

And for instance this `GmailBackupSearchQueryProvider` will call `core.call(....)`
```
 let response = try? core.call("gmailBackupSearch", jsonDict: ["acctEmail": email], data: nil)
 let result = try? response?.json.decodeJson(as: GmailBackupSearchResponse.self)
 return result?.query
```

Reply:
I'm not sure about calling javascript manually this way - I was hoping to contain all javascript-calling things in a single class for ease of maintenance.

If having a `GmailBackupSearchQueryProviderType` helps you with tests, then we can leave it as is.

Reply:
I hope to not mix javascript and swift too much across the codebase - I I'd prefer to contain core calling functionality in core. However, I don't care too much if it's one class for all methods or if it's one class per method, if that helps you somehow.

Reply:
Still `Core` class will handle all javascript logic, but logic of parsing and returning client needed data object will be handled in separate classes. 
But no problems, I will keep it as is.

Reply:
I'd prefer to not do that outside of a contained space (the code is too low level to spread around the app). That could be either one `Core` class, or it could be some sort of `CoreGmailBackupSearch` class, and so on one per method, all in the same folder.